sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -372,4 +385,56 @@ class SidechainApp @Inject()
   def getSecretSubmitProvider: SecretSubmitProvider = secretSubmitProvider
 
   actorSystem.eventStream.publish(SidechainAppEvents.SidechainApplicationStart)
+
+  val timeoutDuration: FiniteDuration = settings.restApi.timeout
+
+  implicit val timeout: Timeout = Timeout(timeoutDuration)
+
+  /***
+   * Retrieve the SidechainBlockId needed to rollback the SidechainStateStorage for the backup.
+   * It's calculated by the following formula:
+   * Genesis_MC_block_height + (current_epch-2) * withdrawalEpochçength -1
+   * @param view
+   * @return sidechainBlockId to rollback or Exception if it's not found
+   */
+  def getSidechainBlockIdForBackUpRollback(view: View):Array[Byte] = {
+    val currentEpoch = view.state.getWithdrawalEpochInfo.epoch
+    val genesisMcBlockHeight = view.history.getMainchainCreationBlockHeight
+    val withdrawalEpochLength = sidechainSettings.genesisData.withdrawalEpochLength
+    val blockHeightToRollback = genesisMcBlockHeight + (currentEpoch -2) * withdrawalEpochLength - 1
+    val mainchainBlockReferenceInfo = view.history.getMainchainBlockReferenceInfoByMainchainBlockHeight(blockHeightToRollback).get()
+    mainchainBlockReferenceInfo.getMainchainReferenceDataSidechainBlockId
+  }
+
+  def createBackup(): Unit = {
+    log.info(s"Starting sidechain backup...")
+
+    val checkAsFuture = (nodeViewHolderRef ? GetDataFromCurrentView(getSidechainBlockIdForBackUpRollback)).asInstanceOf[Future[Array[Byte]]]
+    checkAsFuture.onComplete{
```

Comment:
probably it is an async call. Instead we should lock the threat with `await`

Reply:
Do you mean this GetDataFromCurrentView?

Reply:
I mean `checkAsFuture.onComplete{...}`. So, `SidecahinApp.createBackup()` execution thread will go forward, instead seems better to lock the current thread until we finish the backup. Just for the safety.

Reply:
As you suggested i moved this code inside a REST endpoint, let's see if it's still needed.